pallets/investments/src/lib.rs

```diff
@@ -591,19 +591,22 @@ impl<T: Config> Pallet<T> {
 							Error::<T>::CollectRequired
 						);
 
-						Self::do_update_redeem_order(
-							total_order,
-							&who,
-							investment_id,
-							order,
-							amount,
-						)?;
-
 						order.update_submitted_at(cur_order_id);
 
 						// Remove order from storage if empty
-						if amount == T::Amount::zero() {
+						if amount.is_zero() {
 							*maybe_order = None;
+						} else {
+							// nuno: check that this is ok. Amount == 0 removes the order, so
```

Comment:
Just a note here - `cancel_redeem_order` failed with this change. I reverted it and it worked.

Going to run the other tests now to see if they are affected.

Reply:
> Just a note here - `cancel_redeem_order` failed with this change. I reverted it and it worked.

Had to revert it as well for a bunch of benchmarks in registry and psystem.